"Add new" dialog for all resources

SuaYoo opened the pull request:
Resolves https://github.com/webrecorder/browsertrix-cloud/issues/1197

<!-- Fixes #issue_number -->

### Changes
- Replaces individual "New" buttons in home page with dropdown button in header
- Refactors required step of new workflow and new collection into dialog

### Manual testing

1. Log in. Verify "Add New..." dropdown button is in home dashboard header
2. Click "Add New..." > "Crawl Workflow" > Crawl type. Verify you're taken to the correct form.
3. Click "Start Over". Verify crawl type dialog is shown
4. Go to Crawling page and click "New Workflow". Verify dialog is shown as expected
5. Go to home page
6. Click "Add New..." > "Collection". Verify dialog is shown
7. Enter name and click "Create Collection". Verify you're taken to the crawl selection form.
8. Go to Collections page and click "New Collection". Verify dialog is shown as expected

### Screenshots

| Page | Image/video |
| ---- | ----------- |
| Home | <img width="1132" alt="Screenshot 2023-09-25 at 2 26 15 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/257ca4f0-f8ac-4dde-92ce-de718bef652d"> |
| Home | <img width="191" alt="Screenshot 2023-09-25 at 2 26 19 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/2781134a-6157-4e4a-93ec-be13d7baad75"> |
| Home/Crawling - New Workflow dialog | <img width="765" alt="Screenshot 2023-09-25 at 2 26 26 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/628ada46-14a1-4771-a57f-314f0c60231f"> |
| Home/Collections - New Collection dialog | <img width="752" alt="Screenshot 2023-09-25 at 2 44 37 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/f3549bd6-edef-4cbe-b065-893342d4a50b"> |


<!-- ### Follow-ups -->

SuaYoo committed:
add new dropdown

SuaYoo committed:
move to dialogs

SuaYoo committed:
disable buttons if storage is full

SuaYoo committed:
update titles

SuaYoo committed:
create collection from dialog

SuaYoo committed:
sync url

SuaYoo committed:
adjust header layout

SuaYoo committed:
remove public available switch

Shrinks99 commented:
Yay!  A real action for the overview page!  This is much better than the buttons per section I initially had :)

---

The "Make sharable" switch isn't visible in the "Create a New Collection" dialog.  It should probably be in there!

---

When the New Workflow dialog is triggered from the Crawling page, the contents of the page disappear.  The cursor should also be set to `pointer` on `jobTypeButton`

<img width="1229" alt="Screenshot 2023-09-25 at 9 32 30 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/5672810/ab97a266-75fc-48a8-8164-fc1cbbdd244d">

---

Any strong opinions about the icon button placement?  Personally I think I'd put actual buttons / dropdowns right aligned?

<img width="210" alt="Screenshot 2023-09-25 at 9 20 31 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/5672810/cbed43fc-06db-4bff-a449-dae5b72c6bf8">

---

"Add New" vs "Create New"??  Just making the dropdown "New" doesn't feel good enough, but might keep the language "add" for appending items to a list or similar actions (add exclusion for example)?  Thoughts??

---

SuaYoo commented:
> The "Make sharable" switch isn't visible in the "Create a New Collection" dialog. It should probably be in there!

I hid it since it's not currently working (tracked in https://github.com/webrecorder/browsertrix-cloud/issues/1212) I do wonder in what case someone would want to make a collection without any items shareable? Maybe the shareable toggle should appear elsewhere?

> When the New Workflow dialog is triggered from the Crawling page, the contents of the page disappear. The cursor should also be set to `pointer` on `jobTypeButton`

Nice catch! (edit: Pushed fix)

> Any strong opinions about the icon button placement? Personally I think I'd put actual buttons / dropdowns right aligned?

Switched.

> "Add New" vs "Create New"?? Just making the dropdown "New" doesn't feel good enough, but might keep the language "add" for appending items to a list or similar actions (add exclusion for example)? Thoughts??

No strong feelings for either here, if anyone else wants to weigh in?

SuaYoo committed:
fix workflows list disappearing on new

SuaYoo committed:
switch order of header buttons

SuaYoo committed:
update cursor

tw4l commented:
> > "Add New" vs "Create New"?? Just making the dropdown "New" doesn't feel good enough, but might keep the language "add" for appending items to a list or similar actions (add exclusion for example)? Thoughts??
> 
> No strong feelings for either here, if anyone else wants to weigh in?

Maybe "Create" since that's the verb we use in the dialog?

SuaYoo commented:
> "Add New" vs "Create New"
> Maybe "Create" since that's the verb we use in the dialog?

I think the only one that doesn't completely work is upload, maybe the dropdown item name for that is "Item from File", as in "Create New Item from File"? Or "Add New Uploaded Item"?

Shrinks99 commented:
> I think the only one that doesn't completely work is upload

Bearing in mind that "uploads" are a type of "archived item" I think it makes a reasonable enough amount of sense?  The action of uploading files creates an item?  It's slightly clunky for this instance, but I think it fits better than "add" for all the menu options?

> I do wonder in what case someone would want to make a collection without any items shareable?

I did this the other day!  Had to create a new collection to share some things on our dev instance with one of the folks from Starling.  While creating it, I knew I wanted it to be public so I checked the box on, created the collection, then went _back_ to the Archived Items page to upload a file.  If this dialogue were the only way of creating collections I'd have to re-toggle it elsewhere, adding it right up front saves a step (one less thing to think about as I accomplish other actions!)

Was this the _optimal_ way to accomplish these steps?  Probably not, but it's a good example of providing a consistent interface that serves the users intentions?

Kinda a moot point if it's broken, add it back later!  But I do like having that control available on creation.

SuaYoo committed:
Revert "remove public available switch"

This reverts commit 914f81a3fa22ca51a9f73e9ee635d7cd11e3b62a.

SuaYoo committed:
update isPublic field

SuaYoo committed:
update dropdown button name

SuaYoo commented:
> Bearing in mind that "uploads" are a type of "archived item" I think it makes a reasonable enough amount of sense? 

Good point--updated!

> If this dialogue were the only way of creating collections I'd have to re-toggle it elsewhere, adding it right up front saves a step (one less thing to think about as I accomplish other actions!)

Updated--it looks like https://github.com/webrecorder/browsertrix-cloud/pull/1213 is close to finish, so I added the toggle back.

<img width="759" alt="Screenshot 2023-09-27 at 11 52 22 AM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/cb973080-0a31-4fa9-9103-d1c74ccfc8a6">

SuaYoo committed:
add tooltip

Shrinks99 reviewed (approved):
Looks good!  New tooltip help text is nice!

I also didn't realize this had replaced the create collection action on the collections page too.  Good change!  I like our new unified dialogs :D
